go/src/koding/klient/machine/index/change.go

```diff
@@ -103,6 +103,25 @@ func (cm *ChangeMeta) Coalesce(newer ChangeMeta) ChangeMeta {
 	}
 }
 
+// String implements fmt.Stringer interface and pretty prints stored change.
+func (cm *ChangeMeta) String() string {
+	cpy := atomic.LoadUint64((*uint64)(cm))
+
+	const cmstr = "LRaduH"
```

Comment:
yea this looks a bit long. I was thinking the other way around while creating the map

```
var cmMapping = map[byte]ChangeMeta{
	'L': ChangeMetaLocal,
	'R': ChangeMetaRemote,
	'a': ChangeMetaAdd,
	'd': ChangeMetaRemove,
	'u': ChangeMetaUpdate,
	'H': ChangeMetaLarge,
}

// ...
for c, i := range cmMapping {
	w := getPowerof2(uint64(i))
	if cpy&uint64(i) != 0 {
		buf[w] = c
	} else {
		buf[w] = '-'
	}
}

return string(buf[:len(cmMapping)])
// ...
```

full working example https://play.golang.org/p/cqLp2F8j-O


ps: we can omit `uint64(i)` conversion by putting it into construction of the "cmMapping"